src/test/java/com/example/universe/simulator/entityservice/integration/GalaxyIntegrationTest.java

```diff
@@ -63,14 +64,21 @@ void test() throws Exception {
         resultList = objectMapper.readValue(response.getContentAsString(), new TypeReference<>() {});
         assertThat(resultList.getContent()).hasSize(2);
 
-        // -----------------------------------should return entity-----------------------------------
+        // -----------------------------------should return entity and store it in cache-----------------------------------
 
         // when
         response = performRequest(get("/galaxy/get/{id}", addedDto1.getId()));
         // then
         GalaxyDto resultDto = objectMapper.readValue(response.getContentAsString(), GalaxyDto.class);
         assertThat(resultDto).isEqualTo(addedDto1);
 
+        //check if resultDto got into cache
+        assertThat(redisTemplate.hasKey("Galaxy::" + addedDto1.getId())).isTrue();
+        GalaxyDto fromCache = modelMapper.map(
+                objectMapper.convertValue(redisTemplate.opsForValue().get("Galaxy::" + addedDto1.getId()), Galaxy.class),
+                GalaxyDto.class);
+        assertThat(fromCache).isEqualTo(addedDto1);
```

Comment:
would better to check equality with `resultDto`, not `addedDto1`

Reply:
one should rely on his own test result, not previous execution result, besides diff is clear in next comment also, when you have to increment version.

Reply:
I don't understand the concept of "own test result". First of all this class has only **one** test. I was relying on previous assertion, that's one of the reasons assertions are meant for.
It's fairly simple. If  I know A==B, and I have object C to compare, I can compare it to A , as well as B.

Reply:
here i mean `// when` section, if above some update will be added, your assertion break, that's why closest result should be used